main.go

```diff
@@ -64,13 +65,23 @@ func main() {
 		return
 	}
 
+	// default to config port if the sftp was not passed.
+	if !isFlagPassed("port") {
+		confPort, err := jsonparser.GetInt(config, "sftp", "port")
+		if err != nil {
+			logger.Get().Debugw("could not find sftp port, falling back to \"2022\"", zap.Error(err))
+		}
+		logger.Get().Infow("using config daemon port", zap.String("port", strconv.Itoa(int(confPort))))
+		bindPort = int(confPort)
+	}
+
 	uid, _ := strconv.Atoi(u.Uid)
 	gid, _ := strconv.Atoi(u.Gid)
 
 	var s = server.Configuration{
 		Data:  config,
 		Cache: cache.New(5*time.Minute, 10*time.Minute),
-		User:  server.SftpUser{
+		User: server.SftpUser{
```

Comment:
Your formatter keeps changing things that Goland is unhappy with :(

Reply:
Goland and VSCode have different feeling about formatting...